### ISSUE TITLE
Extending SDRF for immunopeptidomics

### ISSUE THREAD

**Issue by jonasscheid** [bigbio/proteomics-sample-metadata#728]
I think the only thing missing in the current format would be something like `characteristics[mhc type]`. Maybe also `characteristics[mhc class]`

**Comment by TineClaeys** [comment]
I just checked some ontologies and MHC class I and II are mentioned here: https://www.ebi.ac.uk/ols4/ontologies/ncit/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FNCIT_C20706?lang=en
What about adding HLA information, is that necessary for reproducibility?

**Comment by jonasscheid** [comment]
Note for PSI-Meeting: It's also crucial to know the antibody used to pull down the HLA I/II molecule. This can have a large impact which HLA peptides will be identified

**Comment by jonasscheid** [comment]
Nevermind about the normalization of HLA alleles using mhcgnomes. [There are ontologies for 4-digit HLA alleles in OLS, which is nice](https://www.ebi.ac.uk/ols4/ontologies/mro/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMRO_0001454?lang=en) 

The question is: Should we have a column per allele or per locus or per MHC class? How is the structure for multiple variable modifications @TineClaeys ? That needs to a dynamic number of columns right? Per mod one column?

**Comment by mobiusklein** [comment]
This gets even more intricate for class II and if/when we deal with homebrew antibodies. I need to better understand the semantics of rows mapping to samples and raw files because of the way multiple antibodies are multiplexed.

**Comment by jonasscheid** [comment]
> I need to better understand the semantics of rows mapping to samples and raw files because of the way multiple antibodies are multiplexed.
Should be possible to include multiple columns of `comment[antibody purification]` if multiple antibodies are used no?

**Comment by mobiusklein** [comment]
If it's permitted to repeat columns, excellent, let's do that. I know I've seen it done, but I wasn't sure if it was correct or not.

Suppose you have tumor/normal pairs for a sample and the tumor has had loss of one or more HLA alleles. Do we need to explicitly state that or just leave one or more columns null?

**Comment by mobiusklein** [comment]
Reviewing the class II alleles in https://www.ebi.ac.uk/ols4/ontologies/mro/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMRO_0001454?lang=en, they do have some cross-products, but not all allele pairs are crossed. Odds are that the majority of allele pairs we will need are present.

Here are some more stupid questions:

1. When assessing class II typing, it often helps to know whether the class II alleles are being actively expressed in the sample because often with _bulk_ tissue extraction, you don't always get enough antigen presenting cells to matter, and HLA typing is usually done on DNA when using a kit. Do we want to track expression here, or is that too granular a value to report?
2. Do we need to specifically remind the annotator that "if you treated your sample with $\text{IFN}\gamma$ or similar immune agonist, you need to report this with `characteristic[compound]`"?
3. Same for TAP knockout/knockdown?
4. How do you annotate that a cell line was engineered to specifically engineer a specific HLA allele since it will be a newly made cell line? I figure we'll use the parental cell line under `characteristic[cell line]` but we'll need say something specific about the engineered HLA alleles knocked in or out. Alternatively, this could be expressed implicitly in just the listing of `characteristics[mhc type]`

**Comment by jonasscheid** [comment]
>they do have some cross-products, but not all allele pairs are crossed. Odds are that the majority of allele pairs we will need are present.

Agreed, I guess its the best we can get.

Regarding
1. Good point. I think this information is quite valuable _if_ present. That would result in potentially 14 more columns - one expression / quantification value per allele. Although for MHC class II allele combinations, I actually don't know if this is experimentally measurable. I feel like this might explode the metadatasheet quite a bit. Do you think we should not allow it or handle this as _optional_ columns?
2. and 3. Isn't this the factor value for these kind of experiments?
4. For sake of simplicity, I would suggest: `we'll use the parental cell line under characteristic[cell line] + List knock in/out mhcs in characteristics[mhc type]` as you suggested

**Comment by jonasscheid** [comment]
>Going back to the HLA expression question, is it conventional to encode this kind of information in the SDRF? At first it seemed like a reasonable thing to want to know, but on later consideration it seemed like it opens a door to bloating the SDRF with experimental results, not experimental designs.

Agreed, information-gained vs SDRF-complexity-increased ratio is not good imo. Let's leave it out.

How do we continue onwards now?
